<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>compare PGS from a same genotyping event using different imputation panels</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PGS | Analytic validity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Introduction</a>
</li>
<li>
  <a href="0_PredictorPrep.html">Predictors</a>
</li>
<li>
  <a href="1_GRP_ChipData.html">HSU Data</a>
</li>
<li>
  <a href="2_QIMR_ChipData.html">QIMR Data</a>
</li>
<li>
  <a href="3_WGS.html">WGS data</a>
</li>
<li>
  <a href="4_LowPassSeq.html">lcWGS data</a>
</li>
<li>
  <a href="7_Benchmarking.html">Benchmarking</a>
</li>
<li>
  <a href="5_Consistency.html">PGS Consistency</a>
</li>
<li>
  <a href="10_Imputation_Panel_effect.html">Effect of Imputation Panel and missing SNPs</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">compare PGS from a same genotyping event
using different imputation panels</h1>

</div>


<p><br> <br> <br></p>
<p>Four CEPH samples (NA06990, NA07023, NA07057, NA10861) genotyped
multiple times on the same array (Illumina GSAMD-24v1-0_20011747) by
QIMRB were imputed to both HRCr1.1 and TOPMed reference panels.
Comparing the PGS from the two different imputation, we find most traits
highly consistent except a few.</p>
<p>Investigation on the two sets of data suggest the main difference in
PGS are caused by the different SNP coverage. Hereby we developed a new
method to transit the SNP weights to existing SNPs from missing SNPs
when using the high density of SBayesRC predictors.</p>
<div id="summary-difference" class="section level1" number="1">
<h1><span class="header-section-number">1</span> summary difference</h1>
<p>Here is a plot of average PGS comparing between the two imputation
panel.</p>
<pre class="r"><code>mean.prs.summary = melted.control.prs%&gt;% 
  group_by(Study_ID, variable, Impute_Panel) %&gt;%
  summarise(
    mean.prs = mean(value)
  )%&gt;%
  pivot_wider(
    names_from = Impute_Panel,
    values_from = mean.prs
  )

mean.prs.summary$difference = abs(mean.prs.summary$TopMedr3 - mean.prs.summary$HRCr1.1)</code></pre>
<pre class="r"><code>cross.sample.mean = mean.prs.summary %&gt;% 
  group_by(variable) %&gt;% 
  summarise(mean_difference = mean(difference)) %&gt;% 
  arrange(mean_difference)

hist(cross.sample.mean$mean_difference, breaks = 50)</code></pre>
<p><img
src="Figures/difference_of_mean_PGS_between_imputation_panel.png" /></p>
<pre class="r"><code>ggplot(data = mean.prs.summary, aes(x = HRCr1.1, y = TopMedr3, color =  Study_ID)) + geom_point()+ 
  geom_abline(intercept = 0, slope = 1) +
  geom_text_repel(data = mean.prs.summary %&gt;% filter(difference &gt; 0.5 ) , 
                    aes(label = variable), vjust = -1, color = &quot;black&quot;, size = 1.5)</code></pre>
<p><img src="Figures/mean_PGS_comparison_between_imp_panel.jpeg" /></p>
</div>
<div id="pgs-variation-from-imputation-panels" class="section level1"
number="2">
<h1><span class="header-section-number">2</span> PGS Variation from
imputation panels</h1>
<div id="set-up-function-to-plot" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> set up function to
plot</h2>
<pre class="r"><code>## define function
panel_plot = function(example.traits, facet.col, legend.pos, 
                      melted.control.prs, data.vert, melted.prs ){

  ## we already have the data melted.prs, data.vert and melted.control.prs which includes all the traits. 
  ## this function make plots with subsets of traits. 
  
## select traits in data and  simplify to only HRC imputed for control samples
example.melted.prs = melted.prs[which(melted.prs$variable %in% example.traits ),]
example.data.vert = data.vert[which(data.vert$variable %in% example.traits),]
example.melted.control.prs = melted.control.prs[which(melted.control.prs$variable %in% example.traits ),]

# set order
order.matching.table = unique(example.melted.prs [,c(&quot;variable&quot;, &quot;Trait&quot;)])
trait.order.for.plot = order.matching.table[match(example.traits, order.matching.table$variable), &quot;Trait&quot;]


## make them in factor
example.melted.prs$Trait = factor(example.melted.prs$Trait, levels = trait.order.for.plot)
example.data.vert$Trait = factor(example.data.vert$Trait, levels =trait.order.for.plot)
example.melted.control.prs$Trait = factor(example.melted.control.prs$Trait, levels = trait.order.for.plot)

example.melted.control.prs$imputation_panel = factor(example.melted.control.prs$imputation_panel, levels = rev(panel.order) )

xlimleft = round(min(c(-3, min(example.melted.control.prs$value)  ) ) - 0.05 , 1)
xlimright= round(max(c(3, max(example.melted.control.prs$value)  ) ) + 0.05 ,1)

## make plot
example.g.hist = ggplot() + 
  geom_histogram(data =example.melted.prs , aes(x = value ), bins = 150, fill = &quot;light grey&quot;, color = &quot;light grey&quot;) + 
  facet_wrap(~Trait,  ncol= facet.col ) + 
#  geom_vline(data = example.data.vert, mapping = aes(xintercept = NA10861), color = &quot;black&quot;) +
  geom_segment(data = example.data.vert, aes(x =NA10861, xend = NA10861, y= 0, yend = 1100 ) ) +
    geom_point(data = example.melted.control.prs, aes(x = value, y =y.position, color = Study_ID, shape = imputation_panel, alpha = imputation_panel), size = 1) + 
  xlab(&quot;PGS in SD unit&quot;) +
  ylab(&quot;&quot;) + xlim(xlimleft, xlimright) +
    theme_classic(base_size = 14) +
theme(
   panel.grid.major = element_blank(), # Remove major grid lines
    panel.grid.minor = element_blank(), # Remove minor grid lines
    plot.title = element_text(hjust = 0.5),
    axis.title = element_text(),
    axis.text = element_text(size = 12),
    legend.position = legend.pos ,
    legend.text = element_text(size = 12),      # Legend item labels
    legend.title = element_text(size = 14)  ,
    strip.background = element_blank(),
#    strip.text = element_text(margin = margin(b = 20)),  # Add space below facet title
plot.margin = margin(0.1, 0.1, 0.1, 0.1)
   )+
  scale_alpha_manual(name = &quot;Imputation Panel&quot;, values = c(0.4,1) ) +
  scale_shape_manual(name = &quot;Imputation Panel&quot;, values = c(3,2) ) +
  scale_color_manual(
    name = &quot;CEPH Individuals&quot;,
    values = c( color4)) + 
  scale_y_continuous(labels = NULL, breaks = NULL, expand = expansion(mult = c(0, 0.15)))  # Remove both labels and ticks

return(example.g.hist)

}</code></pre>
</div>
<div id="figa.-plot-four-good-examples" class="section level2"
number="2.2">
<h2><span class="header-section-number">2.2</span> figA. plot four good
examples</h2>
<pre class="r"><code>example.traits = c(&quot;Height_03&quot;, &quot;BMI_02&quot;, &quot;Migraine_01&quot;, &quot;IBD_02&quot;, 
                   &quot;Glaucoma_01&quot;, &quot;HairColor_01&quot;, &quot;LungCancer_01&quot;, &quot;UKB_FA_2022_met.d.DHA_pct&quot;)


example.impu.hist1 = panel_plot(example.traits = example.traits[1:4],  facet.col = 4 , legend.pos = &quot;none&quot;,  melted.control.prs = melted.control.prs, data.vert = data.vert, melted.prs =melted.prs )


info.text.a = data.frame(
 variable = example.traits[1:4],
 Trait = predictor.list[match(example.traits[1:4], predictor.list$Predictor),&quot;Label&quot;],
 variance = format(round(predictor.list[match(example.traits[1:4], predictor.list$Predictor),&quot;total.variance&quot;],3 ), nsmall=3) ,
 HRCr1.1 = paste0( round(100 * predictor.list[match(example.traits[1:4], predictor.list$Predictor),&quot;QIMR_HRC&quot;], 1 ) ,  &quot;%&quot;),
 TopMedr3 = paste0( round(100 * predictor.list[match(example.traits[1:4], predictor.list$Predictor),&quot;QIMR_TopMed&quot;], 1 ) ,  &quot;%&quot;),
 x = -0.5,
 y = Inf
)

info.text.a$variable = factor(info.text.a$variable , levels = info.text.a$variable )

info.text.a$Trait = factor(info.text.a$Trait , levels = info.text.a$Trait )

info.text.a$labels = paste0(&quot;V=&quot;, info.text.a$variance ,&quot;; +:&quot;, info.text.a$HRCr1.1, &quot;; \u25B3&quot;,&quot;:&quot; , info.text.a$TopMedr3)

labeled.example.impu.hist1 = example.impu.hist1 + 
  geom_text(
    data = info.text.a,
    aes(x = median(x), y = y, label = (labels) ),
    inherit.aes = FALSE,
    vjust =1,        # Push down from top
    hjust = 0.5,
    size = 4,
    family = &quot;Arial Unicode MS&quot;
  ) 

labeled.example.impu.hist1</code></pre>
<p><img src="Figures/Fig3A_consistent_examples.jpeg" /></p>
</div>
<div id="figb.-four-examples-with-large-difference"
class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> figB. four examples
with large difference</h2>
<pre class="r"><code>example.impu.hist2 = panel_plot(example.traits = example.traits[5:8],  facet.col = 4 , legend.pos = &quot;bottom&quot;, 
                            melted.control.prs = melted.control.prs, data.vert = data.vert, melted.prs =melted.prs )

info.text.b = data.frame(
 variable = example.traits[5:8],
 Trait = predictor.list[match(example.traits[5:8], predictor.list$Predictor),&quot;Label&quot;],
 variance = format(round(predictor.list[match(example.traits[5:8], predictor.list$Predictor),&quot;total.variance&quot;],3 ), nsmall=3) ,
 HRCr1.1 = paste0( round(100 * predictor.list[match(example.traits[5:8], predictor.list$Predictor),&quot;QIMR_HRC&quot;], 1 ) ,  &quot;%&quot;),
 TopMedr3 = paste0( round(100 * predictor.list[match(example.traits[5:8], predictor.list$Predictor),&quot;QIMR_TopMed&quot;], 1 ) ,  &quot;%&quot;),
 x = -0.5,
 y = Inf
)

info.text.b$variable = factor(info.text.b$variable , levels = info.text.b$variable )
info.text.b$Trait = factor(info.text.b$Trait , levels = info.text.b$Trait )

info.text.b$labels = paste0(&quot;V=&quot;, info.text.b$variance ,&quot;; +:&quot;, info.text.b$HRCr1.1, &quot;; \u25B3&quot;,&quot;:&quot; , info.text.b$TopMedr3)

labeled.example.impu.hist2 = example.impu.hist2 + 
  geom_text(
    data = info.text.b,
    aes(x = median(x), y = y, label = (labels) ),
    inherit.aes = FALSE,
    vjust =1,        # Push down from top
    hjust = 0.5,
    size = 4,
    family = &quot;Arial Unicode MS&quot;
  ) 

labeled.example.impu.hist2</code></pre>
<p><img src="Figures/Fig3B_inconsistent_examples.jpeg" /></p>
</div>
</div>
<div id="pgs-variation-of-all-traits-supfig3" class="section level1"
number="3">
<h1><span class="header-section-number">3</span> PGS variation of all
traits (SupFig3 )</h1>
<p>We used the same plot function to display all the 115 traits in 5
groups.</p>
<div id="binary-traits" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> binary traits</h2>
<pre class="r"><code>example.g.hist = panel_plot(example.traits = traits_binary, facet.col = 5, legend.pos = &quot;bottom&quot; , 
                              melted.control.prs = melted.control.prs, data.vert = data.vert, melted.prs =melted.prs)
example.g.hist

fig.heit = ceiling(length(traits_binary) / 5 ) *2
#ggsave(example.g.hist, filename = &quot;Figures/Fig_PGS_benchmarking_example_HRC_vs_TOPmed_binary.jpeg&quot;, width = 12, height = fig.heit)</code></pre>
<p><img
src="Figures/Fig_PGS_benchmarking_example_HRC_vs_TOPmed_binary.jpeg" /></p>
</div>
<div id="quantitative-traits" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> quantitative
traits</h2>
<pre class="r"><code>example.g.hist = panel_plot(example.traits = traits_quanti, facet.col = 5,  legend.pos = &quot;bottom&quot;, 
                            melted.control.prs = melted.control.prs, data.vert = data.vert, melted.prs =melted.prs  )

example.g.hist

fig.heit = ceiling(length( traits_quanti) / 5 ) *2
#fig.heit

#ggsave(example.g.hist ,   filename = &quot;Figures/Fig_PGS_benchmarking_example_HRC_vs_TOPmed_quanti.jpeg&quot;, width = 12, height = fig.heit)</code></pre>
<p><img
src="Figures/Fig_PGS_benchmarking_example_HRC_vs_TOPmed_quanti.jpeg" /></p>
</div>
<div id="fatty-acid" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> fatty acid</h2>
<pre class="r"><code>traits_fa = traits_pfa[grep(&quot;UKB_FA&quot;, traits_pfa)]
example.g.hist = panel_plot(example.traits = traits_fa, facet.col = 5,  legend.pos = &quot;bottom&quot; , 
                            melted.control.prs = melted.control.prs, data.vert = data.vert, melted.prs =melted.prs )

example.g.hist

fig.heit = ceiling(length( traits_fa) / 5 ) *2

#ggsave(example.g.hist ,  filename = &quot;Figures/Fig_PGS_benchmarking_example_HRC_vs_TOPmed_FattyAcid.jpeg&quot;, width = 12, height = 8)</code></pre>
<p><img
src="Figures/Fig_PGS_benchmarking_example_HRC_vs_TOPmed_FattyAcid.jpeg" /></p>
</div>
<div id="proteomics" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> proteomics</h2>
<pre class="r"><code>traits_ppp = traits_pfa[grep(&quot;UKB_PPP&quot;, traits_pfa)]


example.g.hist = panel_plot(example.traits = traits_ppp, facet.col = 5,  legend.pos = &quot;bottom&quot; , 
                            melted.control.prs = melted.control.prs, data.vert = data.vert, melted.prs =melted.prs )

example.g.hist

fig.heit = ceiling(length( traits_ppp) / 5 ) *2

#ggsave(example.g.hist ,  filename = &quot;Figures/Fig_PGS_benchmarking_example_HRC_vs_TOPmed_Proteomics.jpeg&quot;, width = 12, height = 5)</code></pre>
<p><img
src="Figures/Fig_PGS_benchmarking_example_HRC_vs_TOPmed_Proteomics.jpeg" /></p>
</div>
<div id="biomarkers" class="section level2" number="3.5">
<h2><span class="header-section-number">3.5</span> 35 biomarkers</h2>
<pre class="r"><code>example.g.hist = panel_plot(example.traits = traits_35bm, facet.col = 5,  legend.pos = &quot;bottom&quot; , 
                            melted.control.prs = melted.control.prs, data.vert = data.vert, melted.prs =melted.prs )

example.g.hist

fig.heit = ceiling(length( traits_35bm) / 5 ) *2

#ggsave(example.g.hist ,  filename = &quot;Figures/Fig_PGS_benchmarking_example_HRC_vs_TOPmed_35Biomarkers.jpeg&quot;, width = 12, height = fig.heit)</code></pre>
<p><img
src="Figures/Fig_PGS_benchmarking_example_HRC_vs_TOPmed_35Biomarkers.jpeg" /></p>
</div>
</div>
<div id="test-whether-difference-is-from-missing-or-error"
class="section level1" number="4">
<h1><span class="header-section-number">4</span> Test whether difference
is from missing or error</h1>
<div id="profile-pgs-without-the-missing-snps" class="section level2"
number="4.1">
<h2><span class="header-section-number">4.1</span> profile PGS without
the missing SNPs</h2>
<pre class="bash"><code>## define predictor
traitfile=&quot;/QRISdata/Q6913/GCTB_predictor_list_for_batch_profiling.txt&quot;
trait=$(sed &quot;${i}q;d&quot; $traitfile | awk &#39;{print $1}&#39; )
predictor=$(sed  &quot;${i}q;d&quot;  $traitfile  | awk &#39;{print $3}&#39; )
echo $trait
echo $predictor


outdir=PRS_all_GCTB
mkdir -p $outdir

data1=&quot;WGS_NA10861&quot;
genofile1=ceph_combined_SBRC_SNPs.vcf

plink \
  --vcf  $genofile1 \
  --score  $predictor  2 5 8  header sum    \
  --exclude missing_in_any_SNPs_for_NA10861.txt  \
  --out    ${outdir}/${data1}_${trait}_SBayesRC

data2=GSA_HRCr1.1
genofile2=Merged_plink/GSA_HRCr1.1_autosomes

plink \
  --bfile  $genofile2 \
  --score  $predictor  2 5 8  header sum    \
  --exclude missing_in_any_SNPs_for_NA10861.txt  \
  --out    ${outdir}/${data2}_${trait}_SBayesRC

data3=GSA_TopMed
genofile3=Merged_plink/GSA_TOPMedr3_autosomes_7.3M_subset

plink \
  --bfile  $genofile3 \
  --score  $predictor  2 5 8  header sum    \
  --exclude missing_in_any_SNPs_for_NA10861.txt  \
  --out    ${outdir}/${data3}_${trait}_SBayesRC</code></pre>
<p>compare the PGS of NA10861 between the three resources</p>
<pre class="bash"><code>outdir=PRS_all_GCTB
traitfile=&quot;/QRISdata/Q6913/GCTB_predictor_list_for_batch_profiling.txt&quot;

data3=GSA_TopMed
genofile3=Merged_plink/GSA_TOPMedr3_autosomes_7.3M_subset

data2=GSA_HRCr1.1
genofile2=Merged_plink/GSA_HRCr1.1_autosomes

Rscript merge_PGS.R ${genofile2}.fam  $traitfile $outdir $data2
</code></pre>
</div>
<div id="combine-the-data" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> combine the data</h2>
<pre class="r"><code>## input PGS from TopMed imputed
ceph_consis_PRS_TopMed = read.csv(&quot;Data/missingness_check/PRS_all_GCTB_GSA_TopMed_all_139_traits_GCTB_PRS.csv&quot;, row.names =1)
ceph_consis_PRS_TopMed= ceph_consis_PRS_TopMed[,match(included_traits, colnames(ceph_consis_PRS_TopMed) )]
ceph_consis_PRS_TopMed$imputation_panel = &quot;TopMedr3&quot;

## input PGS from HRC imputed
ceph_consis_PRS_HRC = read.csv(&quot;Data/missingness_check/PRS_all_GCTB_GSA_HRCr1.1_all_139_traits_GCTB_PRS.csv&quot;, row.names =1)
ceph_consis_PRS_HRC= ceph_consis_PRS_HRC[,match(included_traits, colnames(ceph_consis_PRS_HRC) )]
ceph_consis_PRS_HRC$imputation_panel = &quot;HRCr1.1&quot;

## merge
ceph_consis_PRS = rbind(ceph_consis_PRS_TopMed, ceph_consis_PRS_HRC)

## get the four samples
pattern &lt;- paste(c(&quot;NA06990&quot;, &quot;NA07023&quot;, &quot;NA07057&quot;, &quot;NA10861&quot;), collapse = &quot;|&quot;)
ceph_consis_PRS &lt;- ceph_consis_PRS[grepl(pattern, row.names(ceph_consis_PRS)), ]


## input PGS from WGS using only overlap SNPs
ceph_consis_PRS_WGS = read.table(&quot;Data/missingness_check/NA10861_WGS_PRS_vGCTB.txt&quot;)
ceph_consis_PRS_WGS$V1 = gsub(&quot;-&quot;, &quot;.&quot;,ceph_consis_PRS_WGS$V1  )
ceph_consis_PRS_WGS = (ceph_consis_PRS_WGS[ match(included_traits, ceph_consis_PRS_WGS$V1),])
colnames(ceph_consis_PRS_WGS) = c(&quot;variable&quot;, &quot;CN1&quot;, &quot;CN2&quot;, &quot;NA10861&quot;)</code></pre>
</div>
<div id="standardize-and-melt" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> standardize and
melt</h2>
<pre class="r"><code>mean.sd.table = read.csv(&quot;Data/LifeLines_MEAN_and_SD_of_traits_for_standardization.csv&quot;, row.names = 1)
mean.sd.table = mean.sd.table[match(included_traits, mean.sd.table$trait ),]

ceph_consis_PRS.norm = sweep(sweep (  sapply (ceph_consis_PRS[,included_traits],  as.numeric  ), 2, mean.sd.table$mean),  2, mean.sd.table$sd, FUN = &#39;/&#39;)
ceph_consis_PRS.norm = data.frame(ceph_consis_PRS.norm)

ceph_consis_PRS.norm$imputation_panel = ceph_consis_PRS$imputation_panel
row.names(ceph_consis_PRS.norm) = row.names(ceph_consis_PRS)


ceph_consis_PRS.norm$IID = sapply(strsplit(row.names(ceph_consis_PRS), &quot;_&quot;), `[`, 4)
ceph_consis_PRS.norm$imputation_panel = factor(ceph_consis_PRS.norm$imputation_panel , levels = panel.order)

ceph_consis_PRS.norm = ceph_consis_PRS.norm %&gt;%  
  arrange(imputation_panel) %&gt;%
  group_by(IID) %&gt;%
  mutate(y.position = row_number()) %&gt;%
  ungroup()  # Ungroup to return to normal data frame structure

  
ceph_consis_PRS.norm$y.position =ceph_consis_PRS.norm$y.position * 5

## melt it
melted_ceph_consis_PRS = reshape2::melt(ceph_consis_PRS.norm, 
                  id.vars =c(&quot;IID&quot;, &quot;imputation_panel&quot;, &quot;y.position&quot;) , 
                  measure.vars =included_traits )
colnames(melted_ceph_consis_PRS)[1] = &quot;Study_ID&quot;

ceph_consis_PRS_WGS$NA10861 =  ( ceph_consis_PRS_WGS$NA10861 -  mean.sd.table$mean) / mean.sd.table$sd

## add label
ceph_consis_PRS_WGS$Trait = predictor.list[match(ceph_consis_PRS_WGS$variable, predictor.list$Predictor),&quot;Label&quot;]
melted_ceph_consis_PRS$Trait =  predictor.list[match(melted_ceph_consis_PRS$variable, predictor.list$Predictor),&quot;Label&quot;]</code></pre>
</div>
<div id="summary-difference-1" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> summary
difference</h2>
<pre class="r"><code>mean.prs.with.missing.summary = melted_ceph_consis_PRS%&gt;% 
  group_by(Study_ID, variable, imputation_panel) %&gt;%
  summarise(
    mean.prs = mean(value)
  )%&gt;%
  pivot_wider(
    names_from = imputation_panel,
    values_from = mean.prs
  )

mean.prs.with.missing.summary$difference = abs(mean.prs.with.missing.summary$TopMedr3 - mean.prs.with.missing.summary$HRCr1.1)

cross.sample.mean.prs.with.missing.summary= mean.prs.with.missing.summary %&gt;% group_by(variable) %&gt;% summarise(mean_difference = mean(difference)) %&gt;% arrange(mean_difference)

cross.sample.mean = data.frame(cross.sample.mean)
cross.sample.mean.prs.with.missing.summary = data.frame(cross.sample.mean.prs.with.missing.summary)
cross.sample.mean$diff_under_same_snps = cross.sample.mean.prs.with.missing.summary[ match(cross.sample.mean$variable, cross.sample.mean.prs.with.missing.summary$variable), &quot;mean_difference&quot;]</code></pre>
<pre class="r"><code>ggplot(data = cross.sample.mean, aes(x = mean_difference, y = diff_under_same_snps)) + geom_point() + geom_abline(intercept = 0, slope =1)</code></pre>
<p><img
src="Figures/mean_difference_of_dif_panel_vs_using_same_snps.jpeg" /></p>
</div>
<div id="sup-fig-6" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> sup fig 6</h2>
<pre class="r"><code>hist.a = ggplot(data = cross.sample.mean, aes(x = mean_difference)) + geom_histogram(bins =50) + xlab( &quot;|PGS-TopMed - PGS-HRC| (in SD unit)&quot;   ) + theme_bw()  + xlim(0,2.2) + ylim(0,55)
hist.b = ggplot(data = cross.sample.mean, aes(x = diff_under_same_snps)) + geom_histogram(bins =50) + xlab( &quot;|PGS-TopMed - PGS-HRC| (in SD unit)&quot;   ) + theme_bw()   + xlim(0,2.2) + ylim(0,55)

hist.supfig6 = ggarrange(hist.a, hist.b, labels = c(&quot;A&quot;, &quot;B&quot;))
hist.supfig6
ggsave( &quot;Figures/SupFig6_hist_of_difference.jpeg&quot;, hist.supfig6, width = 12, height = 5)</code></pre>
<p><img src="Figures/SupFig6_hist_of_difference.jpeg" /></p>
</div>
<div id="plot" class="section level2" number="4.6">
<h2><span class="header-section-number">4.6</span> plot</h2>
<pre class="r"><code>example.traits = example.traits[5:8]


example.consist.hist =  panel_plot(example.traits =example.traits, facet.col = 4,  legend.pos = &quot;bottom&quot; , melted_ceph_consis_PRS , ceph_consis_PRS_WGS, melted.prs )
 

info.text.c = data.frame(
 variable = example.traits,
 Trait = predictor.list[match(example.traits, predictor.list$Predictor),&quot;Label&quot;],
 variance = format(round((predictor.list[match(example.traits, predictor.list$Predictor),&quot;total.variance&quot;]) * (predictor.list[match(example.traits, predictor.list$Predictor),&quot;QIMR_TopMed&quot;]) ,3 ), nsmall=3) ,
# HRCr1.1 = paste0( round(100 * predictor.list[match(example.traits[5:8], predictor.list$Predictor),&quot;QIMR_TopMed&quot;], 1 ) ,  &quot;%&quot;),
# TopMedr3 = paste0( round(100 * predictor.list[match(example.traits[5:8], predictor.list$Predictor),&quot;QIMR_TopMed&quot;], 1 ) ,  &quot;%&quot;),
HRCr1.1 = &quot;100%&quot;,
TopMedr3 = &quot;100%&quot;,
 x = -0.5,
 y = Inf
)

info.text.c$variable = factor(info.text.c$variable , levels = info.text.c$variable )
info.text.c$Trait = factor(info.text.c$Trait , levels = info.text.c$Trait )

info.text.c$labels = paste0(&quot;V=&quot;, info.text.c$variance ,&quot;; +:&quot;, info.text.c$HRCr1.1, &quot;; \u25B3&quot;,&quot;:&quot; , info.text.c$TopMedr3)

labeled.example.consist.hist  = example.consist.hist + 
  geom_text(
    data = info.text.c,
    aes(x = median(x), y = y, label = (labels) ),
    inherit.aes = FALSE,
    vjust =1,        # Push down from top
    hjust = 0.5,
    size = 4,
    family = &quot;Arial Unicode MS&quot;
  ) 

labeled.example.consist.hist</code></pre>
<p><img src="Figures/Fig3C_cons_snp_plot.jpeg" /></p>
</div>
</div>
<div id="examine-convert-function-in-fixing-the-missingness"
class="section level1" number="5">
<h1><span class="header-section-number">5</span> Examine convert
function in fixing the missingness</h1>
<p>We are using correlation of PGS to show how the PGS got affected by
SNP missingness, and how the convert function recover the PGS.</p>
<div id="combine-the-data-1" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> combine the data</h2>
<pre class="r"><code>## input PGS from TopMed imputed using converted predictors
ceph_convert_PRS_TopMed = read.csv(&quot;Data/missingness_check/QIMR_TopMed_PGS_from_converted_predictors_QIMR_GSA_TopMedr3_imputed_all_123_traits_GCTB_PRS.csv&quot;, row.names =1)

included_traits = included_traits[included_traits %in% colnames(ceph_convert_PRS_TopMed)]

ceph_convert_PRS_TopMed= ceph_convert_PRS_TopMed[,match(included_traits, colnames(ceph_convert_PRS_TopMed) )]
ceph_convert_PRS_TopMed$imputation_panel = &quot;TopMedr3&quot;

## input PGS from HRC imputed using original predictor
ceph_PRS_HRC = read.csv(&quot;Data/QIMR/PRS_all_GCTB_GSA_HRCr1.1_all_140_traits_GCTB_PRS.csv&quot;, row.names =1)
ceph_PRS_HRC= ceph_PRS_HRC[,match(included_traits, colnames(ceph_PRS_HRC) )]
ceph_PRS_HRC$imputation_panel = &quot;HRCr1.1&quot;

## merge
ceph_convert_PRS = rbind(ceph_convert_PRS_TopMed, ceph_PRS_HRC)

## get the four samples
pattern &lt;- paste(c(&quot;NA06990&quot;, &quot;NA07023&quot;, &quot;NA07057&quot;, &quot;NA10861&quot;), collapse = &quot;|&quot;)
ceph_convert_PRS &lt;- ceph_convert_PRS[grepl(pattern, row.names(ceph_convert_PRS)), ]


## input PGS from WGS using original predictor
ceph_PRS_WGS = read.table(&quot;Data/WGS/NA10861_WGS_PRS_vGCTB.txt&quot;, header = T)
ceph_PRS_WGS$V1 = gsub(&quot;-&quot;, &quot;.&quot;, ceph_PRS_WGS$V1)
ceph_PRS_WGS = (ceph_PRS_WGS[ match(included_traits, ceph_PRS_WGS$V1),])
colnames(ceph_PRS_WGS) = c(&quot;variable&quot;, &quot;CN1&quot;, &quot;CN2&quot;, &quot;NA10861&quot;)</code></pre>
</div>
<div id="standardize-and-melt-1" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> standardize and
melt</h2>
<pre class="r"><code>mean.sd.table = read.csv(&quot;Data/LifeLines_MEAN_and_SD_of_traits_for_standardization.csv&quot;, row.names = 1)
mean.sd.table = mean.sd.table[match(included_traits, mean.sd.table$trait ),]

ceph_convert_PRS.norm = sweep(sweep (  sapply (ceph_convert_PRS[,included_traits],  as.numeric  ), 2, mean.sd.table$mean),  2, mean.sd.table$sd, FUN = &#39;/&#39;)
ceph_convert_PRS.norm = data.frame(ceph_convert_PRS.norm)

ceph_convert_PRS.norm$imputation_panel = ceph_convert_PRS$imputation_panel
row.names(ceph_convert_PRS.norm) = row.names(ceph_convert_PRS)


ceph_convert_PRS.norm$Study_ID = sapply(strsplit(row.names(ceph_convert_PRS), &quot;_&quot;), `[`, 4)
ceph_convert_PRS.norm$imputation_panel = factor(ceph_convert_PRS.norm$imputation_panel , levels = panel.order)

ceph_convert_PRS.norm = ceph_convert_PRS.norm %&gt;%  
  arrange(imputation_panel) %&gt;%
  group_by(Study_ID) %&gt;%
  mutate(y.position = row_number()) %&gt;%
  ungroup()  # Ungroup to return to normal data frame structure

  
ceph_convert_PRS.norm$y.position =ceph_convert_PRS.norm$y.position * 5

## melt it
melted_ceph_convert_PRS = reshape2::melt(ceph_convert_PRS.norm, 
                  id.vars =c(&quot;Study_ID&quot;, &quot;imputation_panel&quot;, &quot;y.position&quot;) , 
                  measure.vars =included_traits )

ceph_PRS_WGS$NA10861 =  ( ceph_PRS_WGS$NA10861 -  mean.sd.table$mean) / mean.sd.table$sd

## add label

ceph_PRS_WGS$Trait = predictor.list[match(ceph_PRS_WGS$variable, predictor.list$Predictor),&quot;Label&quot;]
melted_ceph_convert_PRS$Trait =  predictor.list[match(melted_ceph_convert_PRS$variable, predictor.list$Predictor),&quot;Label&quot;]</code></pre>
</div>
<div id="summarize" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> summarize</h2>
<pre class="r"><code>mean.prs.with.convert.summary = melted_ceph_convert_PRS%&gt;% 
  group_by(Study_ID, variable, imputation_panel) %&gt;%
  summarise(
    mean.prs = mean(value)
  )%&gt;%
  pivot_wider(
    names_from = imputation_panel,
    values_from = mean.prs
  )

mean.prs.with.convert.summary$difference = abs(mean.prs.with.convert.summary$TopMedr3 - mean.prs.with.convert.summary$HRCr1.1)

cross.sample.mean.prs.with.convert.summary= mean.prs.with.convert.summary %&gt;% group_by(variable) %&gt;% summarise(mean_difference = mean(difference)) %&gt;% arrange(mean_difference)

cross.sample.mean = data.frame(cross.sample.mean)
cross.sample.mean.prs.with.convert.summary = data.frame(cross.sample.mean.prs.with.convert.summary)
cross.sample.mean$diff_from_convert = cross.sample.mean.prs.with.convert.summary[ match(cross.sample.mean$variable, cross.sample.mean.prs.with.convert.summary$variable), &quot;mean_difference&quot;]</code></pre>
</div>
<div id="plot-1" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> plot</h2>
<pre class="r"><code>example.convert.hist =  panel_plot(example.traits =example.traits, facet.col = 4,  legend.pos = &quot;bottom&quot; , melted_ceph_convert_PRS , ceph_PRS_WGS, melted.prs )
 

info.text.d = data.frame(
 variable = example.traits,
 Trait = predictor.list[match(example.traits, predictor.list$Predictor),&quot;Label&quot;],
 variance = format(round((predictor.list[match(example.traits, predictor.list$Predictor),&quot;total.variance&quot;]) ,3 ) , nsmall=3) ,
 #HRCr1.1 = paste0( round(100 * predictor.list[match(example.traits[5:8], predictor.list$Predictor),&quot;QIMR_TopMed&quot;], 1 ) ,  &quot;%&quot;),
 #TopMedr3 = paste0( round(100 * predictor.list[match(example.traits[5:8], predictor.list$Predictor),&quot;QIMR_HRC&quot;], 1 ) ,  &quot;%&quot;),
 HRCr1.1 = &quot;100%&quot;,
TopMedr3 = &quot;100%&quot;,
x = -0.5,
 y = Inf
)

info.text.d$variable = factor(info.text.d$variable , levels = info.text.d$variable )
info.text.d$Trait = factor(info.text.d$Trait , levels = info.text.d$Trait )

info.text.d$labels = paste0(&quot;V=&quot;, info.text.d$variance ,&quot;; +:&quot;, info.text.d$HRCr1.1, &quot;; \u25B3&quot;,&quot;:&quot; , info.text.d$TopMedr3)

labeled.example.convert.hist  = example.convert.hist + 
  geom_text(
    data = info.text.d,
    aes(x = median(x), y = y, label = (labels) ),
    inherit.aes = FALSE,
    vjust =1,        # Push down from top
    hjust = 0.5,
    size = 4,
    family = &quot;Arial Unicode MS&quot;
  ) 

labeled.example.convert.hist

# ggsave(labeled.example.convert.hist, filename = &quot;Figures/Fig4_PGS_consistency_when_using_converted_predictors_in_ToPMed.jpeg&quot;, width = 12, height = 3)</code></pre>
<p><img
src="Figures/Fig3D_PGS_consistency_when_using_converted_predictors_in_ToPMed.jpeg" /></p>
</div>
</div>
<div id="summary-of-changes" class="section level1" number="6">
<h1><span class="header-section-number">6</span> summary of changes</h1>
<pre class="r"><code>cross.sample.mean = cross.sample.mean %&gt;%
    mutate(across(where(is.numeric), ~ round(.x, 3)))

write.csv(cross.sample.mean, &quot;Tables/SupTable3_difference_between_imputation_panel_and_coverted.csv&quot;)

fig.a =  ggplot(data = cross.sample.mean, aes(x = mean_difference, y = diff_from_convert)) + 
  geom_point() + 
  geom_abline(intercept = 0, slope =1)+ 
  geom_text_repel(data = cross.sample.mean %&gt;% filter(mean_difference &gt; 0.2 | diff_from_convert &gt; 0.2) , 
                    aes(label = variable), vjust = -1, color = &quot;black&quot;, size = 1.5)  +
  xlab(&quot;difference between panel&quot;) +
  ylab(&quot;difference after converting the predictor&quot;) +
  theme_bw()

fig.b =  ggplot(data = cross.sample.mean, aes(x = mean_difference, y = diff_under_same_snps)) + 
  geom_point() + 
  geom_abline(intercept = 0, slope =1)+ 
  geom_text_repel(data = cross.sample.mean %&gt;% filter(mean_difference &gt; 0.2 | diff_from_convert &gt; 0.2) , 
                    aes(label = variable), vjust = -1, color = &quot;black&quot;, size = 1.5)  +
  xlab(&quot;difference between panel&quot;) +
  ylab(&quot;difference when using the same SNP set&quot;)+
  theme_bw()

fig.change = ggarrange(fig.b , fig.a,  nrow = 1)



fig.change


ggsave(filename = &quot;Figures/SupFig3_imputation_panel_difference_before_and_after_convert.jpeg&quot;, plot = fig.change, width = 12, height = 6)</code></pre>
<p><img
src="Figures/SupFig3_imputation_panel_difference_before_and_after_convert.jpeg" /></p>
</div>
<div id="check-genotype-of-glaucoma" class="section level1" number="7">
<h1><span class="header-section-number">7</span> check genotype of
Glaucoma</h1>
<p>The top 1000 SNPs were picked from Glaucoma SBsyesRC weights.
Genotype of these 1000 SNPs were extracted from the three data sets that
included sample NA10861: WGS, HRC imputed and TopMed imputed QIMR
Malanoma NCI cohort genotyped with GSAv2 chip.</p>
<p>The genotype value is in additive model as the number of risk
alleles. We removed all the events that have exactly the same genotype
in each data set, which resulted with 1 event from WGS, 1 event from
TopMed imputation, and 4 event from HRC imputation. Then we matched
these 6 events, and removed all the SNPs that have exactly the same
calling among all of them. There are 13 SNPs remained.</p>
<p>Their genotype are merged with the SNP information, and sorted based
on the SBayesRC SNP weight. All the genotype callings that are different
from the calling in WGS are highlighted with red.</p>
<p>We can see that both the count of risk alleles and the partial PGS
calculated from these SNPs are higher in HRC imputed events, which is
consistent with the observation in whole PGS.</p>
<pre class="r"><code>glaucoma.effect = data.frame(fread(&quot;../Large_Data/Glaucoma_01_SBayesRC.snpRes&quot;))
top1000 = read.table(&quot;Data/missingness_check/Glaucoma_01_SBayesRC_top1000_SNPs.txt&quot;)
glaucoma.effect = glaucoma.effect[glaucoma.effect$Name %in% top1000$V1,]


glaucoma.effect$Risk_Allele &lt;- ifelse(glaucoma.effect$A1Effect &gt; 0,
                                      glaucoma.effect$A1,
                                      glaucoma.effect$A2)


glaucoma.effect$Bene_Allele &lt;- ifelse(glaucoma.effect$A1Effect &lt; 0,
                                      glaucoma.effect$A1,
                                      glaucoma.effect$A2)


glaucoma.effect$Risk_Allele_effect = abs(glaucoma.effect$A1Effect)

#write.table(glaucoma.effect[,c(&quot;Name&quot;, &quot;Risk_Allele&quot;)], row.names = F, col.names = F, quote = F, sep =&quot;\t&quot;, file = &quot;Data/missingness_check/top1000SNP_Risk_Allele_of_Glaucoma.txt&quot;)

glaucoma.effect$SNP_Risk_Allele = paste0(glaucoma.effect$Name, &quot;_&quot;, glaucoma.effect$Risk_Allele)
glaucoma.effect$SNP_Bene_Allele = paste0(glaucoma.effect$Name, &quot;_&quot;, glaucoma.effect$Bene_Allele)</code></pre>
<pre class="bash"><code>
## set reference allele
Merged_plink=/QRISdata/Q5740/QIMR_Berghofer/Merged_plink/

plink \
--bfile ${Merged_plink}/GSA_TOPMedr3_autosomes_7.3M_subset \
--extract Glaucoma_top1000_genotype/Glaucoma_01_SBayesRC_top1000_SNPs.txt  \
--exclude missing_in_any_SNPs_for_NA10861.txt  \
--recode A \
--a1-allele top1000SNP_Risk_Allele_of_Glaucoma.txt 1 2 \
--out Glaucoma_top1000_genotype/Glaucoma_01_SBayesRC_top1000_SNPs_in_QIMR_GSA_TOPMedr3 


plink \
--bfile ${Merged_plink}/GSA_HRCr1.1_autosomes \
--extract Glaucoma_top1000_genotype/Glaucoma_01_SBayesRC_top1000_SNPs.txt  \
--exclude missing_in_any_SNPs_for_NA10861.txt  \
--recode A \
--a1-allele top1000SNP_Risk_Allele_of_Glaucoma.txt 1 2 \
--out Glaucoma_top1000_genotype/Glaucoma_01_SBayesRC_top1000_SNPs_in_QIMR_GSA_HRCr1.1 

plink \
--vcf NA10861_combined_SBRC_SNPs.vcf \
--extract Glaucoma_top1000_genotype/Glaucoma_01_SBayesRC_top1000_SNPs.txt  \
--exclude missing_in_any_SNPs_for_NA10861.txt   \
--recode A \
--a1-allele top1000SNP_Risk_Allele_of_Glaucoma.txt 1 2 \
--out Glaucoma_top1000_genotype/Glaucoma_01_SBayesRC_top1000_SNPs_in_WGS_NA10861 
</code></pre>
<pre class="r"><code>geno_trans = function(raw){
# Remove non-SNP columns (e.g., FID, IID, PAT, MAT, SEX, PHENOTYPE)
snp_data &lt;- raw[, 7:ncol(raw)]  # SNP columns start at col 7 in PLINK .raw

# Get individual IDs to set as colnames
individual_ids &lt;- raw$IID

# Transpose
t_snp &lt;- t(snp_data)
colnames(t_snp) &lt;- individual_ids
rownames(t_snp) &lt;- colnames(snp_data)

# Convert to data frame if needed
t_snp_df &lt;- as.data.frame(t_snp)
return(t_snp_df)
}


# Read PLINK raw file
geno.wgs &lt;- read.table(&quot;Data/missingness_check/Glaucoma_01_SBayesRC_top1000_SNPs_in_WGS_NA10861.raw&quot;, header = TRUE)
geno.wgs = geno_trans(geno.wgs)
geno.wgs$toy2 = 1
#geno.wgs$effect_size = &quot;NA&quot;


geno.gsa.topmed = read.table(&quot;Data/missingness_check/Glaucoma_01_SBayesRC_top1000_SNPs_in_QIMR_GSA_TOPMedr3.raw&quot;, header = T)
geno.gsa.topmed = geno_trans(geno.gsa.topmed)
geno.gsa.topmed = geno.gsa.topmed[,grep(&quot;NA10861&quot;, colnames(geno.gsa.topmed))]
geno.gsa.topmed$toy = 1

geno.gsa.hrc = read.table(&quot;Data/missingness_check/Glaucoma_01_SBayesRC_top1000_SNPs_in_QIMR_GSA_HRCr1.1.raw&quot;, header = T)
geno.gsa.hrc = geno_trans(geno.gsa.hrc)
geno.gsa.hrc = geno.gsa.hrc[,grep(&quot;NA10861&quot;, colnames(geno.gsa.hrc))]



# Transpose so samples are rows and SNPs are columns
geno_t &lt;- t(geno.gsa.topmed)
# Find unique rows (i.e., unique samples), then get their indices
unique_indices &lt;- !duplicated(geno_t)
# Subset back to keep only unique columns in the original matrix
geno.gsa.topmed.unique &lt;- geno.gsa.topmed[, unique_indices]
# Optional: check how many duplicates were removed
cat(&quot;Number of duplicate samples removed:&quot;, ncol(geno.gsa.topmed) - ncol(geno.gsa.topmed.unique), &quot;\n&quot;)


# Transpose so samples are rows and SNPs are columns
geno_t &lt;- t(geno.gsa.hrc)
# Find unique rows (i.e., unique samples), then get their indices
unique_indices &lt;- !duplicated(geno_t)
# Subset back to keep only unique columns in the original matrix
geno.gsa.hrc.unique &lt;- geno.gsa.hrc[, unique_indices]
# Optional: check how many duplicates were removed
cat(&quot;Number of duplicate samples removed:&quot;, ncol(geno.gsa.hrc) - ncol(geno.gsa.hrc.unique), &quot;\n&quot;)</code></pre>
<pre class="r"><code>merge_glaucoma_geno &lt;- function(glaucoma.effect, geno) {
  individuals &lt;- colnames(geno)
  output &lt;- matrix(NA, nrow = nrow(glaucoma.effect), ncol = length(individuals))
  rownames(output) &lt;- glaucoma.effect$Name
  colnames(output) &lt;- individuals

  for (i in seq_len(nrow(glaucoma.effect))) {
    snp_a1 &lt;- glaucoma.effect$SNP_Risk_Allele[i]
    snp_a2 &lt;- glaucoma.effect$SNP_Bene_Allele[i]

    if (snp_a1 %in% rownames(geno)) {
      output[i, ] &lt;- as.numeric(geno[snp_a1, , drop = FALSE])
    } else if (snp_a2 %in% rownames(geno)) {
      geno_vals &lt;- as.numeric(geno[snp_a2, , drop = FALSE])
      geno_flipped &lt;- ifelse(geno_vals == 2, 0,
                       ifelse(geno_vals == 0, 2, geno_vals))  # keep 1 as-is
      output[i, ] &lt;- geno_flipped
    }
  }

  geno_df &lt;- as.data.frame(output)
  final &lt;- cbind(glaucoma.effect, geno_df)
  return(final)
}


colnames(geno.gsa.hrc.unique) = paste0(colnames(geno.gsa.hrc.unique), &quot;_HRC&quot;)
colnames(geno.gsa.topmed.unique) = paste0(colnames(geno.gsa.topmed.unique), &quot;_TopMed&quot;)


merged_hrc &lt;- merge_glaucoma_geno(glaucoma.effect, geno.gsa.hrc.unique)

merged_hrc_top  =  merge_glaucoma_geno(merged_hrc, geno.gsa.topmed.unique)
merged_hrc_top_wgs  =  merge_glaucoma_geno(merged_hrc_top, geno.wgs)

merged_hrc_top_wgs=merged_hrc_top_wgs[,grep(&quot;toy&quot;, colnames(merged_hrc_top_wgs), invert = T)]

merged_hrc_top_wgs = merged_hrc_top_wgs %&gt;% arrange(-Risk_Allele_effect)

merged_hrc_top_wgs$Index = 1:1000


last6_cols &lt;- tail(colnames(merged_hrc_top_wgs), 6)

# Keep rows where the values in the last 6 columns are not all the same
filtered_df &lt;- merged_hrc_top_wgs[
  apply(merged_hrc_top_wgs[, last6_cols], 1, function(row) {
    length(unique(row)) &gt; 1
  }),
]

# write.csv(filtered_df, &quot;Tables/SNPs_with_different_geno_in_top1000_of_Glaucoma.csv&quot;, row.names = F)</code></pre>
<pre class="r"><code>filtered_df = read.csv(&quot;Tables/SNPs_with_different_geno_in_top1000_of_Glaucoma.csv&quot;)

colnames(filtered_df) = gsub(&quot;MelanomaMDD_NCI_GSA_&quot; , &quot;&quot;, colnames(filtered_df))

filtered_df  %&gt;%
  select(-Pi1, -Pi2, -Pi3, -Pi4, -Pi5, ) %&gt;% 
  kable() %&gt;%
  kable_styling(full_width = FALSE, bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;)) %&gt;%
  column_spec(17:22, width = &quot;100px&quot;, extra_css = &quot;word-wrap: break-word; white-space: normal;&quot;)</code></pre>
<table class="table table-striped table-hover" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
Index
</th>
<th style="text-align:left;">
Name
</th>
<th style="text-align:right;">
Chrom
</th>
<th style="text-align:right;">
Position
</th>
<th style="text-align:left;">
A1
</th>
<th style="text-align:left;">
A2
</th>
<th style="text-align:right;">
A1Frq
</th>
<th style="text-align:right;">
A1Effect
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
VarExplained
</th>
<th style="text-align:right;">
PIP
</th>
<th style="text-align:left;">
SNP_A1
</th>
<th style="text-align:left;">
SNP_A2
</th>
<th style="text-align:left;">
Risk_Allele
</th>
<th style="text-align:left;">
SNP_Risk_Allele
</th>
<th style="text-align:right;">
Risk_Allele_effect
</th>
<th style="text-align:left;">
Bene_Allele
</th>
<th style="text-align:left;">
SNP_Bene_Allele
</th>
<th style="text-align:right;">
NA10861_WGS
</th>
<th style="text-align:right;">
NA10861_PC08411_TopMed
</th>
<th style="text-align:right;">
NA10861_PC08411_HRC
</th>
<th style="text-align:right;">
NA10861_PC08621_HRC
</th>
<th style="text-align:right;">
NA10861_PC11092_HRC
</th>
<th style="text-align:right;">
NA10861_PC11139_HRC
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
44
</td>
<td style="text-align:left;">
rs4842266
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
79951566
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.696275
</td>
<td style="text-align:right;">
0.034213
</td>
<td style="text-align:right;">
0.033722
</td>
<td style="text-align:right;">
5.44e-05
</td>
<td style="text-align:right;">
0.5307170
</td>
<td style="text-align:left;">
rs4842266_A
</td>
<td style="text-align:left;">
rs4842266_G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
rs4842266_A
</td>
<td style="text-align:right;">
0.034213
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
G
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs4842266_G
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:right;">
117
</td>
<td style="text-align:left;">
rs965652
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
134368953
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.496475
</td>
<td style="text-align:right;">
0.018888
</td>
<td style="text-align:right;">
0.029578
</td>
<td style="text-align:right;">
3.42e-05
</td>
<td style="text-align:right;">
0.3260560
</td>
<td style="text-align:left;">
rs965652_A
</td>
<td style="text-align:left;">
rs965652_G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
rs965652_A
</td>
<td style="text-align:right;">
0.018888
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
G
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs965652_G
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:right;">
131
</td>
<td style="text-align:left;">
rs11755754
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
164325492
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0.399175
</td>
<td style="text-align:right;">
0.017493
</td>
<td style="text-align:right;">
0.025092
</td>
<td style="text-align:right;">
2.52e-05
</td>
<td style="text-align:right;">
0.3695080
</td>
<td style="text-align:left;">
rs11755754_G
</td>
<td style="text-align:left;">
rs11755754_A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
rs11755754_G
</td>
<td style="text-align:right;">
0.017493
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
A
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs11755754_A
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:right;">
151
</td>
<td style="text-align:left;">
rs11614730
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
79925802
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.696100
</td>
<td style="text-align:right;">
0.015212
</td>
<td style="text-align:right;">
0.027911
</td>
<td style="text-align:right;">
2.35e-05
</td>
<td style="text-align:right;">
0.2567350
</td>
<td style="text-align:left;">
rs11614730_A
</td>
<td style="text-align:left;">
rs11614730_G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
rs11614730_A
</td>
<td style="text-align:right;">
0.015212
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
G
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs11614730_G
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:right;">
257
</td>
<td style="text-align:left;">
rs2811687
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
134369822
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.497950
</td>
<td style="text-align:right;">
0.009840
</td>
<td style="text-align:right;">
0.023446
</td>
<td style="text-align:right;">
1.80e-05
</td>
<td style="text-align:right;">
0.1748420
</td>
<td style="text-align:left;">
rs2811687_A
</td>
<td style="text-align:left;">
rs2811687_G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
rs2811687_A
</td>
<td style="text-align:right;">
0.009840
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
G
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs2811687_G
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:right;">
319
</td>
<td style="text-align:left;">
rs2207441
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
164325217
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.401375
</td>
<td style="text-align:right;">
0.008405
</td>
<td style="text-align:right;">
0.018563
</td>
<td style="text-align:right;">
1.10e-05
</td>
<td style="text-align:right;">
0.1442670
</td>
<td style="text-align:left;">
rs2207441_A
</td>
<td style="text-align:left;">
rs2207441_G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
rs2207441_A
</td>
<td style="text-align:right;">
0.008405
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
G
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs2207441_G
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:right;">
354
</td>
<td style="text-align:left;">
rs34852247
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
164324711
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.373500
</td>
<td style="text-align:right;">
0.007586
</td>
<td style="text-align:right;">
0.020255
</td>
<td style="text-align:right;">
1.17e-05
</td>
<td style="text-align:right;">
0.1303810
</td>
<td style="text-align:left;">
rs34852247_A
</td>
<td style="text-align:left;">
rs34852247_G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
rs34852247_A
</td>
<td style="text-align:right;">
0.007586
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
G
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs34852247_G
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:right;">
454
</td>
<td style="text-align:left;">
rs3800902
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
2144378
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0.400550
</td>
<td style="text-align:right;">
-0.006281
</td>
<td style="text-align:right;">
0.019049
</td>
<td style="text-align:right;">
8.30e-06
</td>
<td style="text-align:right;">
0.1571720
</td>
<td style="text-align:left;">
rs3800902_T
</td>
<td style="text-align:left;">
rs3800902_C
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
rs3800902_C
</td>
<td style="text-align:right;">
0.006281
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
T
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs3800902_T
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:right;">
460
</td>
<td style="text-align:left;">
rs34913183
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
58166590
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0.011150
</td>
<td style="text-align:right;">
-0.006232
</td>
<td style="text-align:right;">
0.031416
</td>
<td style="text-align:right;">
1.10e-06
</td>
<td style="text-align:right;">
0.0520962
</td>
<td style="text-align:left;">
rs34913183_T
</td>
<td style="text-align:left;">
rs34913183_C
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
rs34913183_C
</td>
<td style="text-align:right;">
0.006232
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
T
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs34913183_T
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:right;">
776
</td>
<td style="text-align:left;">
rs8016358
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
80895880
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0.427800
</td>
<td style="text-align:right;">
0.004200
</td>
<td style="text-align:right;">
0.012595
</td>
<td style="text-align:right;">
4.80e-06
</td>
<td style="text-align:right;">
0.1101090
</td>
<td style="text-align:left;">
rs8016358_A
</td>
<td style="text-align:left;">
rs8016358_C
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
rs8016358_A
</td>
<td style="text-align:right;">
0.004200
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
C
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs8016358_C
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:right;">
841
</td>
<td style="text-align:left;">
rs4842318
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
80044167
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0.696550
</td>
<td style="text-align:right;">
0.003971
</td>
<td style="text-align:right;">
0.015281
</td>
<td style="text-align:right;">
5.60e-06
</td>
<td style="text-align:right;">
0.0611977
</td>
<td style="text-align:left;">
rs4842318_T
</td>
<td style="text-align:left;">
rs4842318_C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
rs4842318_T
</td>
<td style="text-align:right;">
0.003971
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
C
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs4842318_C
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:right;">
867
</td>
<td style="text-align:left;">
rs8176822
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
80048478
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0.696575
</td>
<td style="text-align:right;">
0.003860
</td>
<td style="text-align:right;">
0.015743
</td>
<td style="text-align:right;">
5.90e-06
</td>
<td style="text-align:right;">
0.0535097
</td>
<td style="text-align:left;">
rs8176822_T
</td>
<td style="text-align:left;">
rs8176822_C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
rs8176822_T
</td>
<td style="text-align:right;">
0.003860
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
C
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs8176822_C
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:right;">
897
</td>
<td style="text-align:left;">
rs2253480
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
6686780
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0.957700
</td>
<td style="text-align:right;">
0.003743
</td>
<td style="text-align:right;">
0.019280
</td>
<td style="text-align:right;">
1.00e-06
</td>
<td style="text-align:right;">
0.0605963
</td>
<td style="text-align:left;">
rs2253480_C
</td>
<td style="text-align:left;">
rs2253480_T
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
rs2253480_C
</td>
<td style="text-align:right;">
0.003743
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
T
</td>
<td style="text-align:left;width: 100px; word-wrap: break-word; white-space: normal;">
rs2253480_T
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
</div>
<div id="check-genotype-of-hair-color" class="section level1"
number="8">
<h1><span class="header-section-number">8</span> check genotype of Hair
color</h1>
<p>The top 1000 SNPs were picked from Hair color SBsyesRC weights.
Genotype of these 1000 SNPs were extracted from the three data sets that
included sample NA10861: WGS, HRC imputed and TopMed imputed QIMR
Malanoma NCI cohort genotyped with GSAv2 chip.</p>
<p>The genotype value is in additive model as the number of risk
alleles. We removed all the events that have exactly the same genotype
in each data set, which resulted with 1 event from WGS, XX event from
TopMed imputation, and XX event from HRC imputation. Then we matched
these 6 events, and removed all the SNPs that have exactly the same
calling among all of them. There are XX SNPs remained.</p>
<p>Their genotype are merged with the SNP information, and sorted based
on the SBayesRC SNP weight. All the genotype callings that are different
from the calling in WGS are highlighted with red.</p>
<p>We can see that both the count of risk alleles and the partial PGS
calculated from these SNPs are higher in HRC imputed events, which is
consistent with the observation in whole PGS.</p>
<pre class="r"><code>## run in bunya
library(data.table)
library(dplyr)
predictor = data.frame(fread(&quot;/QRISdata/Q6913/Uno_Traits/Binary_Traits/HairColor_01/SBayesRC/GCTB_v2.5.2/HairColor_01_SBayesRC.snpRes&quot;))

top1000 &lt;- predictor %&gt;%
  arrange(desc(abs(A1Effect))) %&gt;%
  slice_head(n = 1000)

top1000$Risk_Allele &lt;- ifelse(top1000$A1Effect &gt; 0,
                                      top1000$A1,
                                      top1000$A2)


top1000$Bene_Allele &lt;- ifelse(top1000$A1Effect &lt; 0,
                                      top1000$A1,
                                      top1000$A2)


top1000$Risk_Allele_effect = abs(top1000$A1Effect)

write.table(top1000[,c(&quot;Name&quot;, &quot;Risk_Allele&quot;)], row.names = F, col.names = F, quote = F, sep =&quot;\t&quot;, file = &quot;top1000SNP_Risk_Allele_of_HairColor.txt&quot;)

top1000$SNP_Risk_Allele = paste0(top1000$Name, &quot;_&quot;, top1000$Risk_Allele)
top1000$SNP_Bene_Allele = paste0(top1000$Name, &quot;_&quot;, top1000$Bene_Allele)

write.csv(top1000, &quot;Data/missingness_check/HairColr_top1000.csv&quot;)</code></pre>
<pre class="bash"><code>
## set reference allele
plink=Genotype

plink \
--bfile ${plink}/GSA_TOPMedr3_autosomes_7.3M_subset \
--extract top1000SNP_of_HairColor.txt \
--exclude Lists/missing_in_any_SNPs_for_NA10861.txt  \
--recode A \
--a1-allele top1000SNP_Risk_Allele_of_HairColor.txt 1 2 \
--out HairColor_top1000_SNPs_in_QIMR_GSA_TOPMedr3 


plink \
--bfile ${plink}/GSA_HRCr1.1_autosomes \
--extract top1000SNP_of_HairColor.txt  \
--exclude Lists/missing_in_any_SNPs_for_NA10861.txt  \
--recode A \
--a1-allele top1000SNP_Risk_Allele_of_HairColor.txt 1 2 \
--out HairColor_top1000_SNPs_in_QIMR_GSA_HRCr1.1 

plink \
--vcf ${plink}/NA10861_combined_SBRC_SNPs.vcf \
--extract top1000SNP_of_HairColor.txt  \
--exclude Lists/missing_in_any_SNPs_for_NA10861.txt   \
--recode A \
--a1-allele top1000SNP_Risk_Allele_of_HairColor.txt 1 2 \
--out HairColor_top1000_SNPs_in_WGS_NA10861 
</code></pre>
<pre class="r"><code>geno_trans = function(raw){
# Remove non-SNP columns (e.g., FID, IID, PAT, MAT, SEX, PHENOTYPE)
snp_data &lt;- raw[, 7:ncol(raw)]  # SNP columns start at col 7 in PLINK .raw

# Get individual IDs to set as colnames
individual_ids &lt;- raw$IID

# Transpose
t_snp &lt;- t(snp_data)
colnames(t_snp) &lt;- individual_ids
rownames(t_snp) &lt;- colnames(snp_data)

# Convert to data frame if needed
t_snp_df &lt;- as.data.frame(t_snp)
return(t_snp_df)
}


# Read PLINK raw file
geno.wgs &lt;- read.table(&quot;Data/missingness_check/HairColor_top1000_SNPs_in_WGS_NA10861.raw&quot;, header = TRUE)
geno.wgs = geno_trans(geno.wgs)
geno.wgs$toy2 = 1
#geno.wgs$effect_size = &quot;NA&quot;


geno.gsa.topmed = read.table(&quot;Data/missingness_check/HairColor_top1000_SNPs_in_QIMR_GSA_TOPMedr3.raw&quot;, header = T)
geno.gsa.topmed = geno_trans(geno.gsa.topmed)
geno.gsa.topmed = geno.gsa.topmed[,grep(&quot;NA10861&quot;, colnames(geno.gsa.topmed))]
geno.gsa.topmed$toy = 1

geno.gsa.hrc = read.table(&quot;Data/missingness_check/HairColor_top1000_SNPs_in_QIMR_GSA_HRCr1.1.raw&quot;, header = T)
geno.gsa.hrc = geno_trans(geno.gsa.hrc)
geno.gsa.hrc = geno.gsa.hrc[,grep(&quot;NA10861&quot;, colnames(geno.gsa.hrc))]



# Transpose so samples are rows and SNPs are columns
geno_t &lt;- t(geno.gsa.topmed)
# Find unique rows (i.e., unique samples), then get their indices
unique_indices &lt;- !duplicated(geno_t)
# Subset back to keep only unique columns in the original matrix
geno.gsa.topmed.unique &lt;- geno.gsa.topmed[, unique_indices]
# Optional: check how many duplicates were removed
cat(&quot;Number of duplicate samples removed:&quot;, ncol(geno.gsa.topmed) - ncol(geno.gsa.topmed.unique), &quot;\n&quot;)


# Transpose so samples are rows and SNPs are columns
geno_t &lt;- t(geno.gsa.hrc)
# Find unique rows (i.e., unique samples), then get their indices
unique_indices &lt;- !duplicated(geno_t)
# Subset back to keep only unique columns in the original matrix
geno.gsa.hrc.unique &lt;- geno.gsa.hrc[, unique_indices]
# Optional: check how many duplicates were removed
cat(&quot;Number of duplicate samples removed:&quot;, ncol(geno.gsa.hrc) - ncol(geno.gsa.hrc.unique), &quot;\n&quot;)</code></pre>
<pre class="r"><code>merge_geno &lt;- function(top1000, geno) {
  individuals &lt;- colnames(geno)
  output &lt;- matrix(NA, nrow = nrow(top1000), ncol = length(individuals))
  rownames(output) &lt;- top1000$Name
  colnames(output) &lt;- individuals

  for (i in seq_len(nrow(top1000))) {
    snp_a1 &lt;- top1000$SNP_Risk_Allele[i]
    snp_a2 &lt;- top1000$SNP_Bene_Allele[i]

    if (snp_a1 %in% rownames(geno)) {
      output[i, ] &lt;- as.numeric(geno[snp_a1, , drop = FALSE])
    } else if (snp_a2 %in% rownames(geno)) {
      geno_vals &lt;- as.numeric(geno[snp_a2, , drop = FALSE])
      geno_flipped &lt;- ifelse(geno_vals == 2, 0,
                       ifelse(geno_vals == 0, 2, geno_vals))  # keep 1 as-is
      output[i, ] &lt;- geno_flipped
    }
  }

  geno_df &lt;- as.data.frame(output)
  final &lt;- cbind(top1000, geno_df)
  return(final)
}


top1000 = read.csv(&quot;Data/missingness_check/HairColr_top1000.csv&quot;, row.names =1)

colnames(geno.gsa.hrc.unique) = paste0(colnames(geno.gsa.hrc.unique), &quot;_HRC&quot;)
colnames(geno.gsa.topmed.unique) = paste0(colnames(geno.gsa.topmed.unique), &quot;_TopMed&quot;)


merged_hrc &lt;- merge_geno(top1000, geno.gsa.hrc.unique)

merged_hrc_top  =  merge_geno(merged_hrc, geno.gsa.topmed.unique)
merged_hrc_top_wgs  =  merge_geno(merged_hrc_top, geno.wgs)

merged_hrc_top_wgs=merged_hrc_top_wgs[,grep(&quot;toy&quot;, colnames(merged_hrc_top_wgs), invert = T)]

merged_hrc_top_wgs = merged_hrc_top_wgs %&gt;% arrange(-Risk_Allele_effect)

merged_hrc_top_wgs$Index = 1:1000


last8_cols &lt;- tail(colnames(merged_hrc_top_wgs), 8)

# Keep rows where the values in the last 6 columns are not all the same
filtered_df &lt;- merged_hrc_top_wgs[
  apply(merged_hrc_top_wgs[, last8_cols], 1, function(row) {
    length(unique(row)) &gt; 1
  }),
]

write.csv(filtered_df, &quot;Tables/SNPs_with_different_geno_in_top1000_of_HairColor.csv&quot;, row.names = F)</code></pre>
<pre class="r"><code>filtered_df = read.csv(&quot;Tables/SNPs_with_different_geno_in_top1000_of_HairColor.csv&quot;)

colnames(filtered_df) = gsub(&quot;MelanomaMDD_NCI_GSA_&quot; , &quot;&quot;, colnames(filtered_df))

filtered_df  %&gt;%
  select(-Pi1, -Pi2, -Pi3, -Pi4, -Pi5, ) %&gt;% 
  kable() %&gt;%
  kable_styling(full_width = FALSE, bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;)) %&gt;%
  column_spec(17:22, width = &quot;100px&quot;, extra_css = &quot;word-wrap: break-word; white-space: normal;&quot;)</code></pre>
<table class="table table-striped table-hover" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
Index
</th>
<th style="text-align:left;">
Name
</th>
<th style="text-align:right;">
Chrom
</th>
<th style="text-align:right;">
Position
</th>
<th style="text-align:left;">
A1
</th>
<th style="text-align:left;">
A2
</th>
<th style="text-align:right;">
A1Frq
</th>
<th style="text-align:right;">
A1Effect
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
VarExplained
</th>
<th style="text-align:right;">
PIP
</th>
<th style="text-align:left;">
Risk_Allele
</th>
<th style="text-align:left;">
Bene_Allele
</th>
<th style="text-align:right;">
Risk_Allele_effect
</th>
<th style="text-align:left;">
SNP_Risk_Allele
</th>
<th style="text-align:left;">
SNP_Bene_Allele
</th>
<th style="text-align:right;">
NA10861_PC08411_HRC
</th>
<th style="text-align:right;">
NA10861_PC11071_HRC
</th>
<th style="text-align:right;">
NA10861_PC11075_HRC
</th>
<th style="text-align:right;">
NA10861_PC11129_HRC
</th>
<th style="text-align:right;">
NA10861_PC11139_HRC
</th>
<th style="text-align:right;">
NA10861_PC08411_TopMed
</th>
<th style="text-align:right;">
NA10861_PC11138_TopMed
</th>
<th style="text-align:right;">
NA10861
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
38
</td>
<td style="text-align:left;">
rs10179295
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
222033115
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.405730
</td>
<td style="text-align:right;">
-0.011539
</td>
<td style="text-align:right;">
0.001074
</td>
<td style="text-align:right;">
0.0002685
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0.011539
</td>
<td style="text-align:left;">
rs10179295_G
</td>
<td style="text-align:left;">
rs10179295_A
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:right;">
66
</td>
<td style="text-align:left;">
rs9971729
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
23979791
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0.566359
</td>
<td style="text-align:right;">
-0.004921
</td>
<td style="text-align:right;">
0.002216
</td>
<td style="text-align:right;">
0.0000593
</td>
<td style="text-align:right;">
0.8550120
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0.004921
</td>
<td style="text-align:left;">
rs9971729_A
</td>
<td style="text-align:left;">
rs9971729_C
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:right;">
174
</td>
<td style="text-align:left;">
rs142661815
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
99811314
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.010089
</td>
<td style="text-align:right;">
0.001114
</td>
<td style="text-align:right;">
0.003635
</td>
<td style="text-align:right;">
0.0000012
</td>
<td style="text-align:right;">
0.1107170
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.001114
</td>
<td style="text-align:left;">
rs142661815_A
</td>
<td style="text-align:left;">
rs142661815_G
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:right;">
258
</td>
<td style="text-align:left;">
rs10423391
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
1252451
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0.441128
</td>
<td style="text-align:right;">
0.000749
</td>
<td style="text-align:right;">
0.001361
</td>
<td style="text-align:right;">
0.0000049
</td>
<td style="text-align:right;">
0.2655930
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0.000749
</td>
<td style="text-align:left;">
rs10423391_C
</td>
<td style="text-align:left;">
rs10423391_T
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:right;">
306
</td>
<td style="text-align:left;">
rs10416809
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
1252311
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0.422649
</td>
<td style="text-align:right;">
0.000626
</td>
<td style="text-align:right;">
0.001366
</td>
<td style="text-align:right;">
0.0000046
</td>
<td style="text-align:right;">
0.2211020
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0.000626
</td>
<td style="text-align:left;">
rs10416809_T
</td>
<td style="text-align:left;">
rs10416809_C
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:right;">
334
</td>
<td style="text-align:left;">
rs12609746
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
1252827
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0.241425
</td>
<td style="text-align:right;">
0.000567
</td>
<td style="text-align:right;">
0.001365
</td>
<td style="text-align:right;">
0.0000033
</td>
<td style="text-align:right;">
0.1982280
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0.000567
</td>
<td style="text-align:left;">
rs12609746_C
</td>
<td style="text-align:left;">
rs12609746_T
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:right;">
356
</td>
<td style="text-align:left;">
rs10771034
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
23979199
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0.550739
</td>
<td style="text-align:right;">
-0.000523
</td>
<td style="text-align:right;">
0.001755
</td>
<td style="text-align:right;">
0.0000069
</td>
<td style="text-align:right;">
0.0884168
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0.000523
</td>
<td style="text-align:left;">
rs10771034_T
</td>
<td style="text-align:left;">
rs10771034_A
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:right;">
448
</td>
<td style="text-align:left;">
rs58837341
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
1248920
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.428600
</td>
<td style="text-align:right;">
0.000415
</td>
<td style="text-align:right;">
0.001054
</td>
<td style="text-align:right;">
0.0000026
</td>
<td style="text-align:right;">
0.1678550
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.000415
</td>
<td style="text-align:left;">
rs58837341_T
</td>
<td style="text-align:left;">
rs58837341_G
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:right;">
639
</td>
<td style="text-align:left;">
rs112363642
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
1248348
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.221800
</td>
<td style="text-align:right;">
0.000301
</td>
<td style="text-align:right;">
0.000948
</td>
<td style="text-align:right;">
0.0000014
</td>
<td style="text-align:right;">
0.1306690
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.000301
</td>
<td style="text-align:left;">
rs112363642_A
</td>
<td style="text-align:left;">
rs112363642_G
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:right;">
756
</td>
<td style="text-align:left;">
rs1888379
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
27353599
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0.405515
</td>
<td style="text-align:right;">
0.000263
</td>
<td style="text-align:right;">
0.000973
</td>
<td style="text-align:right;">
0.0000020
</td>
<td style="text-align:right;">
0.1171700
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0.000263
</td>
<td style="text-align:left;">
rs1888379_G
</td>
<td style="text-align:left;">
rs1888379_T
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:right;">
885
</td>
<td style="text-align:left;">
rs16832679
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
181511339
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0.287000
</td>
<td style="text-align:right;">
-0.000232
</td>
<td style="text-align:right;">
0.001151
</td>
<td style="text-align:right;">
0.0000023
</td>
<td style="text-align:right;">
0.0587273
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0.000232
</td>
<td style="text-align:left;">
rs16832679_T
</td>
<td style="text-align:left;">
rs16832679_A
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
1
</td>
<td style="text-align:right;width: 100px; word-wrap: break-word; white-space: normal;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
